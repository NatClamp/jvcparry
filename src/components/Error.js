import React from "react";
import { Div, Icon, Container, Text } from "atomize";

const Loading = () => {
  return (
    <Container>
      <Div d="flex" flexDir='column' align="center" justify="center" p='5rem' h='75vh'>
        <Icon name="Alert" size="80px" />
        <Text tag='h2' textSize='display3'>Something has gone wrong!</Text>
        <Text tag='p' textSize='display1'>Please check back later</Text>
      </Div>
    </Container>
  );
};

export default Loading;
