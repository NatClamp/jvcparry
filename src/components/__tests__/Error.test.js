/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import Error from '../Error';

describe('Error component', () => {
  const history = createMemoryHistory();
  afterEach(cleanup);
  it('should render the error page text', () => {
    render(
      <Router history={history}>
        <Error />
      </Router>,
    );
    expect(screen.getByText(/Something has gone wrong!/i)).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <Error />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
