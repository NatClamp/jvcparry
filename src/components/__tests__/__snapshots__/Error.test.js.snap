// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should match the snapshot 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <svg
      className=""
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      viewBox="0 0 24 24"
    >
      <path
        d="M13.3867068,3.04267403 L22.0176224,18.8660193 C22.5465478,19.835716 22.1892322,21.0505893 21.2195355,21.5795147 C20.9257583,21.7397568 20.5964692,21.8237236 20.2618312,21.8237236 L3,21.8237236 C1.8954305,21.8237236 1,20.9282931 1,19.8237236 C1,19.4890856 1.08396675,19.1597965 1.24420885,18.8660193 L9.87512447,3.04267403 C10.4040499,2.07297735 11.6189232,1.71566168 12.5886199,2.24458714 C12.9257529,2.42847789 13.202816,2.70554099 13.3867068,3.04267403 Z M11.6309156,4.00037829 L3,19.8237236 L20.2618312,19.8237236 L11.6309156,4.00037829 Z M11.6309156,16.3237236 C12.1832004,16.3237236 12.6309156,16.7714388 12.6309156,17.3237236 C12.6309156,17.8760083 12.1832004,18.3237236 11.6309156,18.3237236 C11.0786309,18.3237236 10.6309156,17.8760083 10.6309156,17.3237236 C10.6309156,16.7714388 11.0786309,16.3237236 11.6309156,16.3237236 Z M11.6309156,8.82372359 C12.1832004,8.82372359 12.6309156,9.27143884 12.6309156,9.82372359 L12.6309156,13.8237236 C12.6309156,14.3760083 12.1832004,14.8237236 11.6309156,14.8237236 C11.0786309,14.8237236 10.6309156,14.3760083 10.6309156,13.8237236 L10.6309156,9.82372359 C10.6309156,9.27143884 11.0786309,8.82372359 11.6309156,8.82372359 Z"
        fill="#000000"
        fillRule="evenodd"
      />
    </svg>
    <h2
      className=""
    >
      Something has gone wrong!
    </h2>
    <p
      className=""
    >
      Please check back later
    </p>
  </div>
</div>
`;
